src/auth/src/credentials/external_account_sources/executable_sourced.rs

```diff
@@ -0,0 +1,317 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    Result,
+    constants::{ACCESS_TOKEN_TYPE, JWT_TOKEN_TYPE, SAML2_TOKEN_TYPE},
+    credentials::external_account::SubjectTokenProvider,
+};
+use gax::error::CredentialsError;
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tokio::{process::Command, time::timeout as tokio_timeout};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub(crate) struct ExecutableSourcedCredentials {
+    pub executable: ExecutableConfig,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub(crate) struct ExecutableConfig {
+    pub command: Option<String>,
+    pub timeout_millis: Option<u32>,
+    pub output_file: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+struct ExecutableResponse {
+    version: i32,
+    success: bool,
+    token_type: String,
+    expiration_time: i64,
+    id_token: Option<String>,
+    saml_response: Option<String>,
+    code: Option<String>,
+    message: Option<String>,
+}
+
+impl ExecutableResponse {
+    fn to_cred_error(&self) -> CredentialsError {
+        match &self {
+            ExecutableResponse {
+                message: Some(message),
+                code: Some(code),
+                ..
+            } => {
+                let msg =
+                    format!("{MSG}, response contains unsuccessful response: ({code}) {message}");
+                CredentialsError::from_msg(false, msg)
+            }
+            _ => {
+                let msg = format!(
+                    "{MSG}, response must include `code` and `message` fields when unsuccessful"
+                );
+                CredentialsError::from_msg(false, msg)
+            }
+        }
+    }
+}
+
+const MSG: &str = "failed to read subject token";
+const DEFAULT_TIMEOUT_SECS: u32 = 30;
```

Comment:
You should be able to create a `Duration::from_secs()`, they are `const`:

https://doc.rust-lang.org/std/time/struct.Duration.html#method.from_secs